app/scripts/lib/experiments/base.js

```diff
@@ -0,0 +1,189 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+  An experiment interface that helps organize A/B experiments.
+  It helps keep the state, organize testing groups and log experiment metrics.
+ */
+
+define([
+  'underscore',
+  'lib/storage',
+  'lib/url'
+],
+function (_, Storage, Url) {
+  'use strict';
+
+  var FORCE_GROUP_TYPE = 'forceExperimentGroup';
+  var storage = Storage.factory('localStorage');
+
+  function BaseExperiment() {
+    // nothing to do.
+  }
+
+  _.extend(BaseExperiment.prototype, {}, {
+    /**
+     * Name of the experiment. Used in metrics and view logic.
+     */
+    _name: null,
+
+    /**
+     * The A/B testing group type, see https://en.wikipedia.org/wiki/A/B_testing for details.
+     */
+    _groupType: null,
+
+    /**
+     * Initialize the experiment
+     *
+     * @returns {boolean}
+     *  Returns 'true' when the experiment successfully initialized. Otherwise 'false'.
+     */
+    initialize: function () {
```

Comment:
I'm wondering if initialize/start can be simplified a bit. All subclasses have to call `start` from their `initialize` function. `start` is called with a name, which is always defined within the subclass.

Would it be cleaner to rename the `_name` property to `name`, and each subclass re-define name with the value they pass into start.

`initialize` could call start with the passed in options, and `initialize` within the subclasses could largely go away.